confidence/models.py

```diff
@@ -242,10 +249,19 @@ def __repr__(self):
     __str__ = __repr__
 
 
+# TODO: can't reference NotConfigured yet here, but instantiation needs the value, maybe a cleaner solution is needed?
+_MISSING_MAPPING: typing.MutableMapping[Missing, typing.Any] = {Missing.silent: None,
```

Comment:
What if you re-order these lines a bit, so that you don't have to instantiate this mapping with `Missing.silent: None` first. This would also remove the need for the explicit type hint of the `_MISSING_MAPPING` variable. For example:

```python
NotConfigured = _NotConfigured()
# NB: NotConfigured._missing refers to the NotConfigured *class* at this point, fix this after the name override
NotConfigured._missing = NotConfigured

_MISSING_MAPPING = {Missing.silent: NotConfigured, Missing.error: NoDefault}
```

Reply:
The mapping is needed in `Configuration.__init__`, which is called by `_NotConfigured()`, so it needs to be defined above it.